### ISSUE TITLE
Compiling v.9.2.4 fail

### ISSUE THREAD

**Issue by antonellocaroli** [jcorporation/myMPD#731]
I get this error when I try to compile version 9.2.4

```
[ 99%] Building C object CMakeFiles/mympd.dir/src/lib/cert.c.o
[100%] Linking C executable mympd
/usr/lib/gcc/x86_64-pc-linux-gnu/11.2.1/../../../../x86_64-pc-linux-gnu/bin/ld: CMakeFiles/mympd.dir/dist/rax/rax.c.o: undefined reference to symbol 'log@@GLIBC_2.29'
/usr/lib/gcc/x86_64-pc-linux-gnu/11.2.1/../../../../x86_64-pc-linux-gnu/bin/ld: /lib64/libm.so.6: error adding symbols: DSO missing from command line
collect2: error: ld returned 1 exit status
make[2]: *** [CMakeFiles/mympd.dir/build.make:2054: mympd] Error 1
make[1]: *** [CMakeFiles/Makefile2:100: CMakeFiles/mympd.dir/all] Error 2
make: *** [Makefile:136: all] Error 2
```

**Comment by jcorporation** [comment]
Seems to be a regression of #721. Please provide your distribution, plattform and toolchain.

**Comment by jcorporation** [comment]
Can you apply this diff and test again?
```
diff --git a/CMakeLists.txt b/CMakeLists.txt
index 86f652fc..c5110566 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -93,7 +93,7 @@ endif()
 #required dependencies
 find_package(Threads REQUIRED)
 find_package(PCRE2 REQUIRED)
-find_library(M m REQUIRED)
+find_library(MATH_LIB m REQUIRED)
 
 #optional dependencies
 if(NOT "${ENABLE_SSL}" MATCHES "OFF")
@@ -395,7 +395,7 @@ add_executable(mympd ${MYMPD_SOURCES} ${LIBMPDCLIENT_SOURCES} ${CERT_SOURCES})
 
 #link required dependencies
 target_link_libraries(mympd ${CMAKE_THREAD_LIBS_INIT})
-target_link_libraries(mympd ${M_LIBRARIES})
+target_link_libraries(mympd ${MATH_LIB})
 target_link_libraries(mympd ${PCRE2_LIBRARIES})
 #link optional dependencies
 if(OPENSSL_FOUND)

```

**Comment by antonellocaroli** [comment]
I have two Gentoo systems:

1) does not compile, with the above error

```
app-misc/pax-utils:        1.3.3::gentoo
app-shells/bash:           5.1_p16::gentoo
dev-java/java-config:      2.3.1::gentoo
dev-lang/perl:             5.34.0-r6::gentoo
dev-lang/python:           3.9.10-r1::gentoo
dev-util/cmake:            3.22.2::gentoo
dev-util/meson:            0.60.3::gentoo
sys-apps/baselayout:       2.8::gentoo
sys-apps/openrc:           0.44.10::gentoo
sys-apps/sandbox:          2.29::gentoo
sys-devel/autoconf:        2.71-r1::gentoo
sys-devel/automake:        1.16.5::gentoo
sys-devel/binutils:        2.37_p1-r2::gentoo
sys-devel/binutils-config: 5.4::gentoo
sys-devel/gcc:             11.2.1_p20220115::gentoo
sys-devel/gcc-config:      2.5-r1::gentoo
sys-devel/libtool:         2.4.6-r6::gentoo
sys-devel/make:            4.3::gentoo
sys-kernel/linux-headers:  5.16::gentoo (virtual/os-headers)
sys-libs/glibc:            2.34-r8::gentoo
```

2) compiles without errors

```
app-misc/pax-utils:        1.3.3::gentoo
app-shells/bash:           5.1_p16::gentoo
dev-java/java-config:      2.3.1::gentoo
dev-lang/perl:             5.34.1-r1::gentoo
dev-lang/python:           3.9.12::gentoo
dev-lang/rust-bin:         1.60.0::gentoo
dev-util/cmake:            3.23.1::gentoo
dev-util/meson:            0.61.4-r2::gentoo
sys-apps/baselayout:       2.8::gentoo
sys-apps/openrc:           0.44.10::gentoo
sys-apps/sandbox:          2.29::gentoo
sys-devel/autoconf:        2.71-r1::gentoo
sys-devel/automake:        1.16.5::gentoo
sys-devel/binutils:        2.38-r2::gentoo
sys-devel/binutils-config: 5.4.1::gentoo
sys-devel/gcc:             11.2.1_p20220115::gentoo
sys-devel/gcc-config:      2.5-r1::gentoo
sys-devel/libtool:         2.4.7::gentoo
sys-devel/llvm:            13.0.1::gentoo, 14.0.1::gentoo
sys-devel/make:            4.3::gentoo
sys-kernel/linux-headers:  5.17::gentoo (virtual/os-headers)
sys-libs/glibc:            2.35-r2::gentoo
```

Applying the patch you mentioned compiles on both systems without errors

**Comment by jcorporation** [comment]
Thanks for testing. Is fixed in the v9.3.0 branch and cherry-picked into master.
